### PR TITLE
New serverless pattern - rds-sns-event-notification

### PR DESCRIPTION
**Pull request by jainam1992**
*Issue #1252

*Description of changes:* RDS Event Subscriptions allow users to configure notifications for RDS Events (provided through an SNS topic). This template configures an event subscription for failure, low storage, and availability event categories for RDS Instances.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jbesw** [comment]
The README is not valid. Please update with instructions for this pattern for revalidation.

**Comment by jainam1992** [comment]
@jbesw ReadMe seems correct to me. I am only seeing [this](https://github.com/jainam1992/serverless-patterns/blob/jainam1992-feature-rds-sns-event-notification/rds-sns-event-notification/README.md)

File: cw-sns-secretsmanager/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# AWS Service 1 to AWS Service 2
```
Comment (mavi888) [inline cw-sns-secretsmanager/README.md:1]: Fix the name of the readme file

File: cw-sns-secretsmanager/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+This pattern explains how to deploy a SAM application that Detects and Notify on Amazon Secrets Manager Secret Key Creation, Updation and Deletion using Amazon CloudWatch event and Amazon SNS
+
+Learn more about this pattern at Serverless Land Patterns: << Add the live URL here >>
```
Comment (mavi888) [inline cw-sns-secretsmanager/README.md:5]: add the url to the pattern - https://serverlessland.com/patterns/<folder for the pattern>

File: cw-sns-secretsmanager/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd _patterns-model
```
Comment (mavi888) [inline cw-sns-secretsmanager/README.md:25]: fix

**Comment by mavi888** [comment]
@jainam1992 please fix this if not i will close this PR 

Thanks

**Comment by jainam1992** [comment]
@mavi888 Apologies for the delay. I will fix it soon.

**Comment by jainam1992** [comment]
@mavi888 I have removed some files and did changes as requested. Please let me know if issue still persists. Thank you.

File: rds-sns-event-notification/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# AWS Service 1 to AWS Service 2
```
Comment (mavi888) [inline rds-sns-event-notification/README.md:1]: this is still missing

File: rds-sns-event-notification/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+RDS Event Subscriptions allow users to configure notifications for RDS Events (provided through an SNS topic). This template configures an event subscription for failure, low storage, and availability event categories for RDS Instances.
+
+Learn more about this pattern at Serverless Land Patterns: << Add the live URL here >>
```
Comment (mavi888) [inline rds-sns-event-notification/README.md:5]: add the url to the pattern as instructions in creating the pattern

File: rds-sns-event-notification/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+* [Create an RDS Instance and copy Name of RDS Instance somewhere in notes. You will need it during template deployment] (https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_CreateDBInstance.html#USER_CreateDBInstance.Creating)
```
Comment (mavi888) [inline rds-sns-event-notification/README.md:15]: remove spaces between ]( to make the link appear in the readme

File: rds-sns-event-notification/README.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd _patterns-model
```
Comment (mavi888) [inline rds-sns-event-notification/README.md:25]: fix to the right path

File: rds-sns-event-notification/template.yaml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    Type: "AWS::SNS::TopicPolicy"
+    Properties:
+      PolicyDocument:
+        Statement:
+          - Sid: "__default_statement_ID"
+            Effect: "Allow"
+            Principal:
+              AWS: !Ref 'AWS::AccountId'
+            Action:
+              - "SNS:GetTopicAttributes"
```
Comment (mavi888) [inline rds-sns-event-notification/template.yaml:29]: are all those permissions needed for this to work? What this policy does?

File: rds-sns-event-notification/README.md (hunk 5 above)
Comment (jainam1992) [reply rds-sns-event-notification/README.md]: Done

File: rds-sns-event-notification/README.md (hunk 4 above)
Comment (jainam1992) [reply rds-sns-event-notification/README.md]: Done

File: rds-sns-event-notification/README.md (hunk 6 above)
Comment (jainam1992) [reply rds-sns-event-notification/README.md]: Done

File: rds-sns-event-notification/README.md (hunk 7 above)
Comment (jainam1992) [reply rds-sns-event-notification/README.md]: done

File: rds-sns-event-notification/README.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    sam deploy --guided
+    ```
+1. During the prompts:
+    * Enter a stack name
+    * Enter the desired AWS Region
+    * Allow SAM CLI to create IAM roles with the required permissions.
+    * In SNSEndpoint, Provide your email address to receive notification from Amazon SNS
+    * In RDSInstanceName, Provide name of RDS Instance you created during Deployment instructions
+
+    Once you have run `sam deploy --guided --capabilities CAPABILITY_NAMED_IAM` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
```
[the remark this answers is not in the document]
Comment (jainam1992) [reply rds-sns-event-notification/README.md:38]: sam deploy --guided --capabilities CAPABILITY_NAMED_IAM is correct one. I changed it.

**Comment by jainam1992** [comment]
@mavi888 Fixed all the requested changes. Please review. Thank You.

File: rds-sns-event-notification/src/app.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (jainam1992) [reply rds-sns-event-notification/src/app.js]: deleting file

**Comment by mavi888** [comment]
Hello, this PR is somehow corrupted, the files are good now and everything works, but it doesnt let me merge it. 

Please delete this PR, create a new one - tag me @mavi888 and I will merge it. 

If you dont want to do it I can do it myself but it wont be connected to your username.

**Comment by mavi888** [comment]
This pattern was merged and published in this link: https://serverlessland.com/patterns/rds-sns-event-notification

Thanks a lot for contributing!

**Pull request closed without merging**